static/js/disease/chart.ts

```diff
@@ -17,14 +17,264 @@ import * as d3 from "d3";
 import _ from "lodash";
 const SVGNS = "http://www.w3.org/2000/svg";
 const XLINKNS = "http://www.w3.org/1999/xlink";
+const MARGIN = { top: 30, right: 30, bottom: 90, left: 160 };
+// graph specific dimensions
+const GRAPH_HEIGHT = 400;
+const GRAPH_WIDTH = 760;
+// number to select top data points for large data
+const NUM_DATA_POINTS = 10;
+// number by which y axis domain is increased/decreased for y scale to fit all error bars well
+const Y_AXIS_LIMIT = 0.5;
+// dimension of tissue legend dots
+const LEGEND_CIRCLE_RADIUS = 4;
+// tooltip constant for all charts
+const TOOL_TIP = d3.select("#main").append("div").attr("class", "tooltip");
+// horizontal barcharts - default brightness
+const DEFAULT_BRIGHTEN_PERCENTAGE = "112%";
+// length of the error bar cap for disease-gene associations chart
+const ERROR_BAR_CAP_LENGTH = 10;
+type Datum = DiseaseGeneAssociationData;
+/**
+ * When mouse first enters element specified by given id, brighten it and update/display the global tooltip.
+ */
+function onMouseOver(
+  elementID: string,
+  toolTipText: string,
+  brightenPercentage: string = DEFAULT_BRIGHTEN_PERCENTAGE
+): void {
+  // brighten element: https://stackoverflow.com/a/69610045
+  d3.select(`#${elementID}`).style(
+    "filter",
+    `brightness(${brightenPercentage})`
+  );
+  // update tooltip text
+  TOOL_TIP.html(toolTipText);
+  // show tooltip
+  TOOL_TIP.style("opacity", 1);
+}
+
+/**
+ * Update position of global tooltip to track mouse.
+ */
+function onMouseMove(): void {
+  TOOL_TIP.style("left", d3.event.pageX - 60 + "px").style(
+    "top",
+    d3.event.pageY - 60 + "px"
+  );
+}
+
+/**
+ * When mouse leaves element specified by given id, reset its brightness and hide the global tooltip.
+ */
+function onMouseOut(elementID: string): void {
+  // reset element brightness
+  d3.select(`#${elementID}`).style("filter", "brightness(100%)");
+  // hide tooltip
+  TOOL_TIP.style("opacity", 0);
+}
+
+/**
+ * On mouse hover, select hovered element and
+ *  1) highlight it
+ *  2) update the global tooltip
+ *  3) show the global tooltip.
+ *
+ * Unhighlight and hide the global tooltip when the mouse leaves.
+ */
+function handleMouseEvents(
+  selection: d3.Selection<SVGElement, any, any, any>,
+  idFunc: (index: number) => string,
+  toolTipFunc: (datum: Datum) => string,
+  brightenPercentage: string = DEFAULT_BRIGHTEN_PERCENTAGE
+): void {
+  selection
+    .on("mouseover", (d, i) => {
+      onMouseOver(idFunc(i), toolTipFunc(d), brightenPercentage);
+    })
+    .on("mousemove", onMouseMove)
+    .on("mouseout", (d, i) => onMouseOut(idFunc(i)));
+}
+
+/**
+ * Get function that takes an index and returns an ID containing the chart ID, element name, and index.
+ */
+function getElementIDFunc(
+  chartID: string,
+  elementName: string
+): (index: number) => string {
+  return (index) => `${chartID}-${elementName}${index}`;
+}
+/**
```

Comment:
please ensure an empty line after functions (i imagine the linter should take care of this for you too i.e. `./run_tests.sh -f`).

Reply:
Done!